src/rules/unused_function_params.erl

```diff
@@ -0,0 +1,62 @@
+-module(unused_function_params).
+
+-behaviour(hank_rule).
+
+-export([analyze/1]).
+
+-spec analyze(hank_rule:asts()) -> [hank_rule:result()].
+analyze(ASTs) ->
+    lists:foldl(fun({File, AST}, Acc) ->
+                   FunctionsTrees = extract_functions_from_ast(AST),
+                   ct:print("~p~n", [FunctionsTrees]),
+                   Acc ++ check_functions(FunctionsTrees)
+                end,
+                [],
+                ASTs).
+
+check_functions(FunctionsTrees) ->
+    [check_function(FunctionTree) || FunctionTree <- FunctionsTrees].
+
+check_function(FunctionTree) ->
+    ClauseTrees = erl_syntax:function_clauses(FunctionTree),
+    % [{tree,clause,
+    %    {attr,[{text,"multi_fun"},{location,10}],[],none},
+    %    {clause,[{atom,[{text,"undefined"},{location,10}],undefined},
+    %             {var,[{text,"_Arg2"},{location,10}],'_Arg2'}],
+    %            none,
+    %            [{atom,[{text,"ok"},{location,11}],ok}]}},
+    % [(_a, b, _c), (_x, b, _c)]
+    % [{1, 0, 1}, {1, 0, 1}] warning!
+    %
+    % [(a, _b, c), (_x, b, c)]
+    % [{0, 1, 0}, {1, 0, 0}] ok
```

Comment:
You shouldn't rely on internal structures. You should either use `erl_syntax` functions everywhere …or maybe use `ktn_code` like `elvis_core` does.